openlcs/libs/corgi.py

```diff
@@ -597,7 +597,11 @@ def should_yield_data(component, result):
                 try:
                     component = next(components)
                 except StopIteration:
-                    yield result
+                    # StopIteration may appear when `should_yield_data` is
+                    # False, i.e., the last iteration(before exhausting) may
+                    # accumulates components less than `num_components`.
+                    if len(result) > 1:
+                        yield result
```

Comment:
there're cases when len(result["sources"]) < `num_components` which keeps it in the dict till `StopIteration` is raised, needs to be yielded anyway.

One corner case is by the time the `StopIteration` occurs, the `result["sources"]` is empty, which everything's already yielded, thus we simply break it, and nothing else needs to be done.